azurerm/resource_arm_policy_set_definition.go

```diff
@@ -0,0 +1,283 @@
+package azurerm
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"reflect"
+	"strings"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+
+	"time"
+
+	"strconv"
+
+	"encoding/json"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/policy"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/structure"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmPolicySetDefinition() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmPolicySetDefinitionCreateUpdate,
+		Update: resourceArmPolicySetDefinitionCreateUpdate,
+		Read:   resourceArmPolicySetDefinitionRead,
+		Delete: resourceArmPolicySetDefinitionDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			"policy_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(policy.TypeBuiltIn),
+					string(policy.TypeCustom),
+					string(policy.TypeNotSpecified),
+				}, true)},
```

Comment:
since this is a new field, can we make this case sensitive here?
```suggestion
				}, false)},
```

Reply:
is there an example of when `NotSpecified` would be useful? I have a feeling we may not want to expose that?